### PR TITLE
fixed build with nginx/1.11.11

### PR DESCRIPTION
**Pull request by defanator**
Closes openresty/echo-nginx-module#64

See also:
http://hg.nginx.org/nginx/rev/e662cbf1b932

### CODE REVIEW

File: src/ngx_http_echo_request_info.c [hunk 1]
```diff
@@ -215,8 +218,15 @@ ngx_http_echo_client_request_headers_variable(ngx_http_request_t *r,
 
     if (hc->nbusy) {
         b = NULL;
+
+#if defined(nginx_version) && nginx_version >= 1011011
+         for (cl = hc->busy; cl; /* void */) {
```
Comment (agentzh) [inline src/ngx_http_echo_request_info.c]: Bad indentation? We should multiples of 4 spaces here.

**Comment by agentzh** [comment]
@dyu Have you got the whole test suite pass on our side with nginx 1.11.11+?

**Comment by agentzh** [comment]
@defanator Your patch breaks a lot of tests in the `t/request-info.t` test file of this module's test suite when using nginx 1.11.11:

https://gist.github.com/agentzh/4e749e224eae2eaba0ee397e384deaf3

**Comment by defanator** [comment]
@agentzh, 24923db fixes the `request-info.t`.

I have to say that I'm getting other tests failed on both nginx/1.11.10 and nginx/1.11.11, in particular:

```
Test Summary Report
-------------------
t/echo-timer.t      (Wstat: 256 Tests: 14 Failed: 1)
  Failed test:  10
  Non-zero exit status: 1
t/location.t        (Wstat: 256 Tests: 96 Failed: 1)
  Failed test:  12
  Non-zero exit status: 1
t/subrequest.t      (Wstat: 768 Tests: 130 Failed: 3)
  Failed tests:  34, 36, 64
  Non-zero exit status: 3
Files=23, Tests=1118, 48 wallclock secs ( 0.23 usr  0.05 sys +  4.54 cusr  1.66 csys =  6.48 CPU)
Result: FAIL
```

However, those failures are sporadical and do not seem to be related to the proposed change.

**Comment by agentzh** [comment]
@defanator Right. I had noted those as well.

**Comment by agentzh** [comment]
@defanator Merged with the following patch:

```diff
diff --git a/.travis.yml b/.travis.yml
index 2c563527..7534c3c6 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -50,7 +50,7 @@ env:
     - TEST_NGINX_SLEEP=0.006
   matrix:
     - NGINX_VERSION=1.9.15
-    - NGINX_VERSION=1.11.11
+    - NGINX_VERSION=1.11.2
 
 services:
  - memcache
diff --git a/src/ngx_http_lua_common.h b/src/ngx_http_lua_common.h
index f2cfbe8a..f37d776a 100644
--- a/src/ngx_http_lua_common.h
+++ b/src/ngx_http_lua_common.h
@@ -200,8 +200,9 @@ struct ngx_http_lua_main_conf_s {
     ngx_uint_t           malloc_trim_req_count;
 
 #if nginx_version >= 1011011
-    ngx_buf_t          **busy_bufs_ptrs;
-    ngx_int_t            prealloc_nbusy;
+    /* the following 2 fields are only used by ngx.req.raw_headers() for now */
+    ngx_buf_t          **busy_buf_ptrs;
+    ngx_int_t            busy_buf_ptr_count;
 #endif
 
     unsigned             requires_header_filter:1;
diff --git a/src/ngx_http_lua_headers.c b/src/ngx_http_lua_headers.c
index 986f06eb..6700ce80 100644
--- a/src/ngx_http_lua_headers.c
+++ b/src/ngx_http_lua_headers.c
@@ -164,22 +164,22 @@ ngx_http_lua_ngx_req_raw_header(lua_State *L)
 
     if (hc->nbusy) {
 #if nginx_version >= 1011011
-        if (hc->nbusy > lmcf->prealloc_nbusy) {
-            if (lmcf->busy_bufs_ptrs) {
-                ngx_free(lmcf->busy_bufs_ptrs);
+        if (hc->nbusy > lmcf->busy_buf_ptr_count) {
+            if (lmcf->busy_buf_ptrs) {
+                ngx_free(lmcf->busy_buf_ptrs);
             }
 
-            lmcf->busy_bufs_ptrs = ngx_alloc(hc->nbusy * sizeof(ngx_buf_t *),
-                                             r->connection->log);
+            lmcf->busy_buf_ptrs = ngx_alloc(hc->nbusy * sizeof(ngx_buf_t *),
+                                            r->connection->log);
 
-            if (lmcf->busy_bufs_ptrs == NULL) {
+            if (lmcf->busy_buf_ptrs == NULL) {
                 return luaL_error(L, "no memory");
             }
 
-            lmcf->prealloc_nbusy = hc->nbusy;
+            lmcf->busy_buf_ptr_count = hc->nbusy;
         }
 
-        bb = lmcf->busy_bufs_ptrs;
+        bb = lmcf->busy_buf_ptrs;
         for (cl = hc->busy; cl; cl = cl->next) {
             *bb++ = cl->buf;
         }
@@ -187,7 +187,7 @@ ngx_http_lua_ngx_req_raw_header(lua_State *L)
         b = NULL;
 
 #if nginx_version >= 1011011
-        bb = lmcf->busy_bufs_ptrs;
+        bb = lmcf->busy_buf_ptrs;
         for (i = hc->nbusy; i > 0; i--) {
             b = bb[i - 1];
 #else
@@ -270,9 +270,9 @@ ngx_http_lua_ngx_req_raw_header(lua_State *L)
     if (hc->nbusy) {
 
 #if nginx_version >= 1011011
-        bb = lmcf->busy_bufs_ptrs;
-        for (i = hc->nbusy; i > 0; i--) {
-            b = bb[i - 1];
+        bb = lmcf->busy_buf_ptrs;
+        for (i = hc->nbusy - 1; i >= 0; i--) {
+            b = bb[i];
 #else
         for (i = 0; i < hc->nbusy; i++) {
             b = hc->busy[i];
@@ -1491,9 +1491,9 @@ ngx_http_lua_ngx_raw_header_cleanup(void *data)
 
     lmcf = (ngx_http_lua_main_conf_t *) data;
 
-    if (lmcf->busy_bufs_ptrs) {
-        ngx_free(lmcf->busy_bufs_ptrs);
-        lmcf->busy_bufs_ptrs = NULL;
+    if (lmcf->busy_buf_ptrs) {
+        ngx_free(lmcf->busy_buf_ptrs);
+        lmcf->busy_buf_ptrs = NULL;
     }
 }
 #endif

```

**Comment by agentzh** [comment]
Note: nginx 1.11.11+ are still not an officially supported target of this module yet.

More work is needed. For example, I'm still seeing (at least) the following test failures when using nginx 1.11.11 though they are not related to this PR:

```
t/045-ngx-var.t ..................... 1/54
#   Failed test 'TEST 8: $proxy_host & $proxy_port - raw resp headers like'
#   at /home/agentzh/git/lua-nginx-module/../test-nginx/lib/Test/Nginx/Socket.pm line 950.
#                   'HTTP/1.1 200 OK
# Server: nginx/1.11.11 (no pool)
# Date: Thu, 13 Apr 2017 21:59:35 GMT
# Content-Type: text/plain
# Transfer-Encoding: chunked
# Connection: close
# '
#     doesn't match '(?^s:Proxy-Host: 127.0.0.1\:\d+\r
# Proxy-Port: \d+\r
# )'

#   Failed test 'TEST 8: $proxy_host & $proxy_port - raw resp headers like'
#   at /home/agentzh/git/lua-nginx-module/../test-nginx/lib/Test/Nginx/Socket.pm line 950.
#                   'HTTP/1.1 200 OK
# Server: nginx/1.11.11 (no pool)
# Date: Thu, 13 Apr 2017 21:59:35 GMT
# Content-Type: text/plain
# Transfer-Encoding: chunked
# Connection: close
# '
#     doesn't match '(?^s:Proxy-Host: 127.0.0.1\:\d+\r
# Proxy-Port: \d+\r
# )'
# Looks like you failed 2 tests of 54.
t/045-ngx-var.t ..................... Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/54 subtests
```

**Comment by agentzh** [comment]
Oops, sorry, commented on the wrong project. My bad.

**Comment by CamJN** [comment]
Why has this been left unmerged? I'm considering using the patch, but perhaps the maintainers have reason to to accept it as is?

**Comment by agentzh** [comment]
@defanator Merged with minor edits. Thanks!

**Pull request closed without merging**